open_instruct/actor_manager.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2024 The AllenAI Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ActorManager for controlling evaluation and weight updates across all LLMRayActors."""
+
+import collections
+import socket
+import threading
+import time
+from datetime import datetime
+from pathlib import Path
+
+import uvicorn
+from fastapi import FastAPI
+from fastapi.responses import HTMLResponse
+from fastapi.staticfiles import StaticFiles
+
+from open_instruct import logger_utils
+
+
+def find_free_port():
+    """Find and return a free port number."""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
+        s.listen(1)
+        port = s.getsockname()[1]
+    return port
+
+
+class ActorManager:
+    """Centralized manager for controlling evaluation and weight updates across all LLMRayActors."""
+
+    def __init__(self, queues: dict, args, enable_dashboard: bool = True):
+        self._should_stop = False
+        self._last_updated = datetime.now()
+        self._dashboard_port = None
```

Comment:
should we allow the end user to specify a dashboard port optionally? not a strong feeling but maybe could be useful?

Reply:
I'm happy to add that but we should disable it on beaker, as there's no nice way to get a port allocated in experiment mode (not supported). So you have to use the port finding code. Do you still think it's worth adding given that?